### PR TITLE
Use ledger action to make ledger more reliable

### PR DESCRIPTION
**Pull request by NullSoldier**
## Summary

This UI action will handle many of the failure cases that occur when
running ledger commands. They'll use CLI-UX to inform the user of the
current state, and what the state should be.

TODO:
- [x] Make it work seemlessly when existing actions are running such as when using the interactive multisig server
- [x] Investigate issue with lock screen spamming

## Testing Plan

- Plug in a ledger and perform multisig with a server and without a server.
- Plug in a ledger and perform multisig while hosting the server
- Plug in a ledger and perform multisig while joining the server

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and label it with `breaking-change-rpc` or `breaking-change-sdk`.

```
[ ] Yes
```

### CODE REVIEW

**Comment by patnir** [comment]
Tested this with a bunch of cases - device falls asleep, i lock the device, and the app is not open. 

image

File: ironfish-cli/src/commands/wallet/multisig/dkg/create.ts [hunk 1]
```diff
@@ -70,15 +70,7 @@ export class DkgCreateCommand extends IronfishCommand {
...
       ledger = new LedgerMultiSigner(this.logger)
-      try {
-        await ledger.connect()
-      } catch (e) {
-        if (e instanceof Error) {
-          this.error(e.message)
-        } else {
-          throw e
-        }
-      }
```
Comment (patnir) [inline ironfish-cli/src/commands/wallet/multisig/dkg/create.ts:81]: This happens in a lot of places, needs to be replaced everywhere. Can happen later though

File: ironfish-cli/src/ui/ledger.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+          ux.action.start('Ledger handshake failed... trying again')
+        } else if (e instanceof LedgerClaNotSupportedError) {
+          const appName = ledger.isMultisig ? 'Ironfish' : 'Ironfish DKG'
+          ux.action.start(`Wrong Ledger app opened. Please open ${appName} APP.`)
+        } else if (e instanceof LedgerAppNotOpen) {
+          const appName = ledger.isMultisig ? 'Ironfish' : 'Ironfish DKG'
+          ux.action.start(`Unlock your Ledger and open the ${appName} APP.`)
+        } else {
+          throw e
+        }
```
Comment (patnir) [inline ironfish-cli/src/ui/ledger.ts:48]: seeing this work is so sick. assuming once this PR gets in we need to use ledgerAction everywhere?

**Comment by patnir** [comment]
This is draft, but I would approve.

**Review by hughy — APPROVED** [review]
Tested out and works well!

I think as a future improvement we might move some log output out of the Ledger methods so that it doesn't repeat on retry. For example:
image

**Pull request merged**